real_decoder.go

```diff
@@ -290,6 +290,14 @@ func (rd *realDecoder) peek(offset, length int) (packetDecoder, error) {
 	return &realDecoder{raw: rd.raw[off : off+length]}, nil
 }
 
+func (rd *realDecoder) peekInt8(offset int) (int8, error) {
+	if rd.remaining() < offset+1 {
+		return -1, ErrInsufficientData
+	}
+	tmp := int8(rd.raw[rd.off+offset])
```

Comment:
you can avoid tmp variable by - 

```suggestion
	return int8(rd.raw[rd.off+offset]), nil
```

Reply:
tmp is not giving much info anyway, if you want a variable, please make it bit descriptive.